core/dbt/clients/registry.py

```diff
@@ -31,18 +36,34 @@ def _get(path, registry_base_url=None):
     fire_event(RegistryProgressGETResponse(url=url, resp_code=resp.status_code))
     resp.raise_for_status()
 
-    # It is unexpected for the content of the response to be None so if it is, raising this error
-    # will cause this function to retry (if called within _get_with_retries) and hopefully get
-    # a response.  This seems to happen when there's an issue with the Hub.
+    # It is unexpected for the content of the response to be None or malformed.  If it is,
+    # raising this error will cause this function to retry (if called within _get_with_retries)
+    # and hopefully get a valid response.  This seems to happen when there's an issue with the Hub.
     # See https://github.com/dbt-labs/dbt-core/issues/4577
-    if resp.json() is None:
+    # and https://github.com/dbt-labs/dbt-core/issues/4849
+    if not validJson(resp.json()):
         raise requests.exceptions.ContentDecodingError(
-            "Request error: The response is None", response=resp
+            "Request error: The response is not valid json", response=resp
         )
     return resp.json()
 
 
+def validJson(jsonData):
+    try:
+        # the quotes in the response were converted to single quotes which is not
+        # valid json.  json.dumps() will fix that.
+        json.loads(json.dumps(jsonData))
```

Comment:
This feels bad.  Any suggestions on a better way to do this welcome!

Reply:
The comment doesn't state this yet but the `api/v1/index.json` path also returns a list objects that causes it fail.